test/integration/scripting_spec.mjs

```diff
@@ -2367,4 +2367,41 @@ describe("Interaction", () => {
       );
     });
   });
+
+  describe("Textfield with a zip code starting with 0", () => {
+    let pages;
+    let otherPages;
+
+    beforeAll(async () => {
+      otherPages = await Promise.all(
+        global.integrationSessions.map(async session =>
+          session.browser.newPage()
+        )
+      );
+      pages = await loadAndWait("bug1889122.pdf", getSelector("24R"));
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+      await Promise.all(otherPages.map(page => page.close()));
+    });
+
+    it("must check the zip code is correctly formatted", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page], i) => {
+          await page.waitForFunction(
+            "window.PDFViewerApplication.scriptingReady === true"
+          );
+
+          await page.click(getSelector("24R"));
+          await page.type(getSelector("24R"), "01234", { delay: 10 });
+          await page.keyboard.press("Tab");
+          await waitForTimeout(100);
```

Comment:
Given that we should eventually get rid of `waitForTimeout`, see issue #17656, we should probably avoid adding new occurrences of it unless absolutely necessary. Would it be possible in this case to simply use `waitForSelector` to wait until the selector has the value we want it to have, since that's also the pattern we used in the previous occurrences where waiting for an arbitrary amount of time was already removed?